File: ads/2023_1/POO/aula02_08-03-2023/Atividades/atividade03/ContaBancaria.cs
namespace Atividade03{
    public class ContaBancaria{
        public double Saldo { get; private set; }
        public int Numero { get; set;}

        public ContaBancaria(int numero){
            Numero = numero;
        }

        public void Depositar(double valor){
            Saldo += valor;
            Console.WriteLine($"Depósito de R${valor} efetuado / Saldo atual: {Saldo}");
        }

        public bool Sacar(double valor){
            if (Saldo >= valor) {
                Saldo -= valor;
                Console.WriteLine($"Saque de R${valor} efetuado / Saldo atual: {Saldo}");
                return true;
            } else {
                Console.WriteLine($"Saldo insuficiente.");
                return false;
            }
        }
    }
}

File: ads/2023_1/Laboratório de Programação/aula06_31-03-2023/atividade de reforco/atividade04/src/ContaBancaria.java
public class ContaBancaria {
    public int numeroDaConta;
    public double saldo;
    public String titular;

    public void depositar(double valor) {
        saldo += valor;
        System.out.println("Depósito de R$" + valor + " realizado com sucesso. Novo saldo: R$" + saldo);
    }

    public void sacar(double valor) {
        if (saldo >= valor) {
            saldo -= valor;
            System.out.println("Saque de R$" + valor + " realizado com sucesso. Novo saldo: R$" + saldo);
        } else {
            System.out.println("Saldo insuficiente para realizar o saque.");
        }
    }

    public void transferir(double valor, ContaBancaria destino) {
        if (saldo >= valor) {
            saldo -= valor;
            destino.saldo += valor;
            System.out.println("Transferência de R$" + valor + " realizada com sucesso para " + destino.titular + ". Novo saldo: R$" + saldo);
        } else {
            System.out.println("Saldo insuficiente para realizar a transferência.");
        }
    }
}
